x-pack/plugins/apm/server/routes/transaction_groups.ts

```diff
@@ -75,23 +76,35 @@ export const transactionGroupsChartsRoute = createRoute(() => ({
       transactionType,
       transactionName,
       uiFilters: uiFiltersJson,
+      environment,
     } = context.params.query;
 
-    const uiFilters = getParsedUiFilters({ uiFilters: uiFiltersJson, logger });
-
     const searchAggregatedTransactions = await getSearchAggregatedTransactions(
       setup
     );
 
-    return getTransactionCharts({
+    const options = {
       serviceName,
       transactionType,
       transactionName,
       setup,
       searchAggregatedTransactions,
       logger,
-      uiFilters,
-    });
+      environment,
+    };
+
+    if (uiFiltersJson) {
+      const uiFilters = getParsedUiFilters({
+        uiFilters: uiFiltersJson,
+        logger,
+      });
+      return getTransactionCharts({
+        ...options,
+        uiFilters,
+      });
+    }
+
+    return getTransactionCharts(options);
```

Comment:
This doesn't feel right, and perhaps points to another problem. 
Currently `setup` exposes `uiFiltersES` but not `uiFilters`. I think the following should be renamed:

 - `getUiFiltersES` -> `getEsFilter`
 - `uiFiltersES` -> `esFilter`

And then setup can expose both `esFilter` (type: `ESFilter[]`) and `uiFilters` (type: `UIFilters`):

```ts
const uiFilters = uiFiltersEncoded ? JSON.parse(uiFiltersEncoded) : undefined;
const esFilter = uiFilters ? getEsFilter(uiFilters) : [];

return {
  uiFilters,
  esFilter,
  start: '...',
  end: '...',
...coreSetupRequest,
}
```

Then we can get rid of the following methods:
 -  [`getParsedUiFilters`](https://github.com/elastic/kibana/blob/ac8b8516a654c72eeeff421def5b9e05af01eb3e/x-pack/plugins/apm/server/lib/helpers/convert_ui_filters/get_parsed_ui_filters.ts#L10)
 - [`decodeUiFilters`](https://github.com/elastic/kibana/blob/ac8b8516a654c72eeeff421def5b9e05af01eb3e/x-pack/plugins/apm/server/lib/helpers/setup_request.ts#L29-L35)

Reply:
I agree it's more clear the way you propose. This PR has definitely surfaced a fair bit of tech debt 😄

Reply:
Yeah, sorry for pulling you into it. You okay with doing it as part of this PR?